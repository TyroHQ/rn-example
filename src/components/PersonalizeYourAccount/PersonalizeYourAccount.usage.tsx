import React from "react";
import { boolean, text, number } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";
import { PersonalizeYourAccount } from "./PersonalizeYourAccount";

export const basic = () => <PersonalizeYourAccount />;
