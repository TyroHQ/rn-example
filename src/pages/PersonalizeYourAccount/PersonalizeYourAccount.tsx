import React from "react";
import { Text, View } from "react-native";
import styled from "styled-components";
import { Placeholder } from "@src/components/Placeholder";

const PersonalizeYourAccountContainer = styled(View)`
  background-color: lightgrey;
  padding: 10px;
`;

export class PersonalizeYourAccount extends React.Component<{}> {
  static defaultProps: PersonalizeYourAccount["props"];
  render() {
    return (
      <Placeholder>
        <Placeholder>Header bar</Placeholder>
        <Placeholder>
          Content area
          <Placeholder>Header text</Placeholder>
          <Placeholder>Explainer text</Placeholder>
          <Placeholder>User icon</Placeholder>
          <Placeholder>Explainer text</Placeholder>
          <Placeholder>
            Input fields
            <Placeholder>Input textfield</Placeholder>
            <Placeholder>Input textfield</Placeholder>
            <Placeholder>Input textfield</Placeholder>
          </Placeholder>
          <Placeholder>Explainer text</Placeholder>
          <Placeholder>
            Radio buttons
            <Placeholder>
              Radio button1
              <Placeholder>Selected radiobox</Placeholder>
              <Placeholder>Input placeholder</Placeholder>
            </Placeholder>
            <Placeholder>
              Radio button2
              <Placeholder>Unselected radiobox</Placeholder>
              <Placeholder>Input placeholder</Placeholder>
            </Placeholder>
          </Placeholder>
          <Placeholder>
            Rectangular buttons
            <Placeholder>Rectangular buttonleft</Placeholder>
            <Placeholder>Rectangular buttonright</Placeholder>
          </Placeholder>
        </Placeholder>
      </Placeholder>
    );
  }
}
