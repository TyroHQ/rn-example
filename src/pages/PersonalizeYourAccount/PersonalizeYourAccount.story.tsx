import { storiesOf } from "@storybook/react-native";
import figmaDecorator from "storybook-addon-figma";

import * as usage from "./PersonalizeYourAccount.usage";

let figma = undefined;

let story = storiesOf("PersonalizeYourAccount", module);

if (figma) {
  story = story.addDecorator(
    figmaDecorator({
      url: figma
    })
  );
}

for (const example in usage) {
  story = story.add(example, (usage as any)[example]);
}
