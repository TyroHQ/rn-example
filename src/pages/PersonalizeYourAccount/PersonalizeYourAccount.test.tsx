import { shallow } from "enzyme";
import * as usage from "./PersonalizeYourAccount.usage";

describe("<PersonalizeYourAccount />", () => {
  for (const example in usage) {
    it(`renders well in ${example} mode`, () => {
      const x = (usage as any)[example]();
      const wrapper = shallow(x);
      expect(wrapper).toMatchSnapshot();
    });
  }
});
