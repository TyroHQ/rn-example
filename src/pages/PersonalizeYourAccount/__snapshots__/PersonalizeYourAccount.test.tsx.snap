// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PersonalizeYourAccount /> renders well in basic mode 1`] = `
<Placeholder
  orientation="row"
>
  <Placeholder
    orientation="row"
  >
    Header bar
  </Placeholder>
  <Placeholder
    orientation="row"
  >
    Content area
    <Placeholder
      orientation="row"
    >
      Header text
    </Placeholder>
    <Placeholder
      orientation="row"
    >
      Explainer text
    </Placeholder>
    <Placeholder
      orientation="row"
    >
      User icon
    </Placeholder>
    <Placeholder
      orientation="row"
    >
      Explainer text
    </Placeholder>
    <Placeholder
      orientation="row"
    >
      Input fields
      <Placeholder
        orientation="row"
      >
        Input textfield
      </Placeholder>
      <Placeholder
        orientation="row"
      >
        Input textfield
      </Placeholder>
      <Placeholder
        orientation="row"
      >
        Input textfield
      </Placeholder>
    </Placeholder>
    <Placeholder
      orientation="row"
    >
      Explainer text
    </Placeholder>
    <Placeholder
      orientation="row"
    >
      Radio buttons
      <Placeholder
        orientation="row"
      >
        Radio button1
        <Placeholder
          orientation="row"
        >
          Selected radiobox
        </Placeholder>
        <Placeholder
          orientation="row"
        >
          Input placeholder
        </Placeholder>
      </Placeholder>
      <Placeholder
        orientation="row"
      >
        Radio button2
        <Placeholder
          orientation="row"
        >
          Unselected radiobox
        </Placeholder>
        <Placeholder
          orientation="row"
        >
          Input placeholder
        </Placeholder>
      </Placeholder>
    </Placeholder>
    <Placeholder
      orientation="row"
    >
      Rectangular buttons
      <Placeholder
        orientation="row"
      >
        Rectangular buttonleft
      </Placeholder>
      <Placeholder
        orientation="row"
      >
        Rectangular buttonright
      </Placeholder>
    </Placeholder>
  </Placeholder>
</Placeholder>
`;
